source/portable/NetworkInterface/libslirp/MBuffNetworkInterface.c

```diff
@@ -77,76 +77,86 @@ extern void vMBuffNetifBackendDeInit( void * pvBackendContext );
 
 static void vNetifReceiveTask( void * pvParameters );
 
-MBuffNetDriverContext_t xDriverCtx = { 0 };
-
 /**
  * @brief Initialize the MessageBuffer backed network interface.
  *
  * @return BaseType_t pdTRUE on success
  */
-BaseType_t xNetworkInterfaceInitialise( void )
+BaseType_t xNetworkInterfaceInitialise( NetworkInterface_t * pxNetif )
 {
     BaseType_t xResult = pdTRUE;
 
-    if( xDriverCtx.xInterfaceState == pdFALSE )
+    pxNetif->pvArgument = pvPortMalloc( sizeof( MBuffNetDriverContext_t ) );
```

Comment:
@ChaiTowKway, I am rescinding my approval because there is no check for the case where malloc fails.

Reply:
configASSERT() check has been added for if malloc fails per other malloc calls in the project:
Example in [BufferAllocation_2.c](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/source/portable/BufferManagement/BufferAllocation_2.c#L200), and [WinPCap/NetworkInterface.c](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/source/portable/NetworkInterface/WinPCap/NetworkInterface.c#L954)